client/ayon_core/pipeline/editorial.py

```diff
@@ -419,17 +419,50 @@ def get_media_range_with_retimes(otio_clip, handle_start, handle_end):
     else:
         # compute retimed range
         media_in_trimmed = conformed_source_range.start_time.value + offset_in
-        media_out_trimmed = media_in_trimmed + (
-            (
-                conformed_source_range.duration.value
-                * abs(time_scalar)
-                + offset_out
-            ) - 1
-        )
+
+        offset_duration = conformed_source_range.duration.value * abs(time_scalar)
+        offset_duration -= 1  # duration 1 frame -> freeze frame -> end = start + 0
+        offset_duration = max(0, offset_duration)  # negative duration = frozen frame
+        media_out_trimmed = media_in_trimmed + offset_duration
 
         media_in = available_range.start_time.value
         media_out = available_range.end_time_inclusive().value
 
+    if time_warp_nodes:
+        # Naive approach: Resolve consecutive timewarp(s) on range,
+        # then check if plate range has to be extended beyond source range.
+        frame_range = [media_in_trimmed]
+        in_frame = media_in_trimmed + time_scalar
+        while in_frame <= media_out_trimmed:
+            frame_range.append(in_frame)
+            in_frame += time_scalar
```

Comment:
If `time_scalar` is `int` then this can be simplified to:
```suggestion
        frame_range = list(range(
            media_in_trimmed,
            media_out_trimmed + 1,
            time_scalar
        ))
```

Reply:
time_scalar might be `float` (unfortunately 😞)